OctoAwesome/OctoAwesome.Basics/EntityComponents/ToolBarComponent.cs

```diff
@@ -91,5 +83,40 @@ public void AddNewSlot(InventorySlot slot)
                 }
             }
         }
+        public override void Update(GameTime gameTime, IGameService service)
+        {
+            IEntityController controller = (Entity as IControllable)?.Controller;
+            if (controller == null) return;
+            else if(controller.InteractInput)
+            {
+                if (Entity.Components.TryGetComponent(out InventoryComponent inventory))
+                {
+                    service.TakeBlock(controller, Entity.Cache, inventory);
+                }
+                else if (service.TakeBlock(controller, Entity.Cache, out IInventoryableDefinition item))
+                {
+                    // TODO: und jetzt ?
```

Comment:
Ich finde es etwas komisch, dass die ToolbarComponent direkt mit der Welt interagiert.

Reply:
Warum nicht ? Ob es diese Komponente macht oder eine andere ist für mich kein Unterschied. Die Komponenten berechnen sich selbst wenn es geht. Dann braucht man kein typeof(T) gesuche mehr in irgendwelchen Liste.